pkg/pod/status.go

```diff
@@ -127,6 +127,13 @@ func MakeTaskRunStatus(ctx context.Context, logger *zap.SugaredLogger, tr v1.Tas
 
 	complete := areContainersCompleted(ctx, pod) || isPodCompleted(pod)
 
+	// When EnableKubernetesSidecar is true, we need to ensure all init containers
+	// are completed before considering the taskRun complete, in addition to the regular containers.
+	// This is because sidecars in Kubernetes can keep running after the main containers complete.
+	if config.FromContextOrDefaults(ctx).FeatureFlags.EnableKubernetesSidecar {
```

Comment:
NIT: I wonder if this could be checked in all cases (even if sidecar is not enabled). 
That said, having the check helps reducing the scope of the patch and makes it a better candidate for backporting.